cmds/copy.c

```diff
@@ -29,42 +29,40 @@ static void cmd_copyInfo(void)
 
 static ssize_t cmd_cpphfs2phfs(handler_t srcHandler, addr_t srcAddr, size_t srcSz, handler_t dstHandler, addr_t dstAddr, size_t dstSz)
 {
-	int res;
-	addr_t addr = dstAddr;
-	size_t size, chunk;
+	ssize_t res;
 	u8 buff[SIZE_MSG_BUFF];
+	size_t chunk, rsz = 0, wsz = 0;
 
 	/* Size is not defined, copy the whole file                 */
 	if (srcSz == 0 && dstSz == 0)
-		size = -1;
+		srcSz = -1;
 	/* Size is defined, use smaller one to copy piece of memory */
 	else if (srcSz != 0 && dstSz != 0)
-		size = (srcSz < dstSz) ? srcSz : dstSz;
+		srcSz = (srcSz < dstSz) ? srcSz : dstSz;
 	/* One of the size is not defined, use the defined one      */
 	else
-		size = (srcSz > dstSz) ? srcSz : dstSz;
+		srcSz = (srcSz > dstSz) ? srcSz : dstSz;
 
 	do {
-		if (size > sizeof(buff))
-			chunk = sizeof(buff);
-		else
-			chunk = size;
-
-		if ((res = phfs_read(srcHandler, srcAddr, buff, chunk)) < 0) {
+		chunk = ((srcSz - rsz) > sizeof(buff)) ? sizeof(buff) : srcSz;
```

Comment:
Should be:
`chunk = ((srcSz - rsz) > sizeof(buff)) ? sizeof(buff) : (srcSz - rsz);`
Right now the last read will have size of all others + `(srcSz - rsz)`;

Reply:
Yeah, you have right. I overlooked it. Fixed